src/main/src/main/java/org/geoserver/ows/LanguageURLMangler.java

```diff
@@ -20,24 +19,13 @@ public void mangleURL(
             StringBuilder baseURL, StringBuilder path, Map<String, String> kvp, URLType type) {
         if (Dispatcher.REQUEST.get() == null || !type.equals(URLType.SERVICE)) return;
 
-        String languageParameter = null;
-        String acceptLanguagesParameter = null;
-
-        HttpServletRequest httpRequest = Dispatcher.REQUEST.get().getHttpRequest();
-        Enumeration<String> parameterNames = httpRequest.getParameterNames();
-
-        for (String parameter : Collections.list(parameterNames)) {
-            if (parameter.equalsIgnoreCase(LANGUAGE)) {
-                languageParameter = parameter;
-            } else if (parameter.equalsIgnoreCase(ACCEPT_LANGUAGES)) {
-                acceptLanguagesParameter = parameter;
-            }
-        }
+        final Map<String, Object> rawKvp =
+                Optional.ofNullable(Dispatcher.REQUEST.get().rawKvp).orElse(new KvpMap<>());
 
         String language = "";
 
-        if (acceptLanguagesParameter != null) {
-            String acceptLanguages = httpRequest.getParameter(acceptLanguagesParameter);
+        if (rawKvp.containsKey(ACCEPT_LANGUAGES)) {
+            String acceptLanguages = (String) rawKvp.get(ACCEPT_LANGUAGES);
             if (acceptLanguages != null) {
```

Comment:
I would check here if the string is not empty, or even if the string as a comma.